eos-chrome-plugin-update-upstream

```diff
@@ -34,104 +34,119 @@ WIDEVINE_VERSION=
-        || fp_exit_retry_with_error "Failed to download Packages from Google Chrome"
+fp_fetch_widevine_metadata() {
+    if ! ${WGET_COMMAND} "${CHROME_PACKAGES}" -O "${TEMP_DIR}/Packages"; then
+        echo "Failed to download Packages from Google Chrome"
+        return 1
+    fi
+
     local package=$(fp_get_field "Package")
-    [ -n "${package}" ] || \
-        fp_exit_with_error "Packages does not contain ${CHROME_FLAVOUR_VERSION}"
+    if [ -z "${package}" ]; then
+        echo "Packages does not contain ${CHROME_FLAVOUR_VERSION}"
+        return 1
+    fi
+
     CHROME_VERSION=$(fp_get_field "Version")
     CHROME_SHA1SUM=$(fp_get_field "SHA1")
+
     local filename=$(fp_get_field "Filename")
     CHROME_URL_FILENAME=${CHROME_URL}/${filename}
     CHROME_FILENAME=${CHROME_URL_FILENAME##*/}
 }
 
-fp_should_check_update() {
-    if [ ! -f ${WIDEVINE_VERSION_FILE} ] ; then
-        WIDEVINE_VERSION="not installed"
-    elif [ `find ${WIDEVINE_VERSION_FILE} -daystart -mtime +${DAYS_CHECK}` ] || \
-        [ `find ${WIDEVINE_VERSION_FILE} -newermt "0 days"` ] ; then
-        WIDEVINE_VERSION=$(cat ${WIDEVINE_VERSION_FILE})
+fp_should_update_widevine() {
+    if fp_fetch_widevine_metadata; then
+       if [ "${WIDEVINE_VERSION}" = "${CHROME_VERSION}" ] ; then
+           # Touch the VERSION so it is not checked again until ${DAYS_CHECK} passed
+           echo "Widevine plugin is up-to-date"
+           touch ${WIDEVINE_VERSION_FILE}
+           return 1
+       else
+           echo "A new version of the widevine plugin is available: ${CHROME_VERSION}"
+           rm -f ${WIDEVINE_VERSION_FILE}
+           return 0
+       fi
     else
-        fp_exit "No need to check yet"
-    fi
-}
-
-fp_should_update() {
-    if [ "${WIDEVINE_VERSION}" = "${CHROME_VERSION}" ] ; then
-        # Touch the VERSION so it is not checked again until ${DAYS_CHECK} passed
-        touch ${WIDEVINE_VERSION_FILE}
-        fp_exit "Widevine plugin is up-to-date"
+        echo "An error happened fetching metadata for the Widevine plugin"
+        return 1
     fi
 }
 
-fp_download_chrome() {
+fp_download_google_chrome() {
     # Get Chrome
     echo "Downloading ${CHROME_FILENAME}"
-    ${WGET_COMMAND} "${CHROME_URL_FILENAME}" -O "${TEMP_DIR}/${CHROME_FILENAME}" \
-        || fp_exit_retry_with_error "Failed to download ${CHROME_FILENAME}"
-
-    # Verify SHA1 checksum of debian file
-    echo "Verifying ${TEMP_DIR}/${CHROME_FILENAME}"
-
-    echo "${CHROME_SHA1SUM}  ${TEMP_DIR}/${CHROME_FILENAME}" | sha1sum -c > /dev/null 2>&1 \
-	|| fp_exit_with_error "sha1sum mismatch ${TEMP_DIR}/${CHROME_FILENAME}"
+    if ${WGET_COMMAND} "${CHROME_URL_FILENAME}" -O "${TEMP_DIR}/${CHROME_FILENAME}"; then
+        # Verify SHA1 checksum of debian file
+        echo "Verifying ${TEMP_DIR}/${CHROME_FILENAME}"
+        echo "${CHROME_SHA1SUM} ${TEMP_DIR}/${CHROME_FILENAME}" | sha1sum -c > /dev/null 2>&1
+        if [ $? != 0 ]; then
```

Comment:
Not sure if this works as you expect... the script runs with `bash -e` and will automatically exit whenever an instruction returns a status that is non-zero.
You should however be able to enclose that under an `if` condition like you did for the `dpkg` command below.